src/content/capture/video-contenthint/js/main.js

```diff
@@ -0,0 +1,151 @@
+/*
+*  Copyright (c) 2016 The WebRTC project authors. All Rights Reserved.
+*
+*  Use of this source code is governed by a BSD-style license
+*  that can be found in the LICENSE file in the root of the source
+*  tree.
+*/
+
+'use strict';
+
+var srcVideo = document.getElementById('srcVideo');
+var fluidVideo = document.getElementById('fluidVideo');
+var detailedVideo = document.getElementById('detailedVideo');
+
+var srcStream;
+var fluidStream;
+var detailedStream;
+
+var offerOptions = {
+  offerToReceiveAudio: 0,
+  offerToReceiveVideo: 1
+};
+
+function maybeCreateStream() {
+  if (srcStream) {
+    return;
+  }
+  if (srcVideo.captureStream) {
+    srcStream = srcVideo.captureStream();
+    console.log('Captured stream from srcVideo with captureStream',
+      srcStream);
+    call();
+  } else if (srcVideo.mozCaptureStream) {
+    srcStream = srcVideo.mozCaptureStream();
+    console.log('Captured stream from srcVideo with mozCaptureStream()',
+      srcStream);
+    call();
+  } else {
+    trace('captureStream() not supported');
+  }
+}
+
+// Video tag capture must be set up after video tracks are enumerated.
+srcVideo.oncanplay = maybeCreateStream;
+if (srcVideo.readyState >= 3) {  // HAVE_FUTURE_DATA
+  // Video is already ready to play, call maybeCreateStream in case oncanplay
+  // fired before we registered the event handler.
+  maybeCreateStream();
+}
+
+srcVideo.play();
+
+function setVideoTrackContentHints(stream, hint) {
+  var tracks = stream.getVideoTracks();
+  tracks.forEach(function(track) {
+    if ('contentHint' in track) {
+      track.contentHint = hint;
+      if (track.contentHint !== hint) {
+        trace('Invalid video track contentHint: \'' + hint + '\'');
+      }
+    } else {
+      trace('MediaStreamTrack contentHint attribute not supported');
+    }
+  });
+}
+
+function call() {
+  fluidStream = srcStream.clone();
+  setVideoTrackContentHints(fluidStream, 'fluid');
+  establishPC(fluidVideo, fluidStream);
```

Comment:
This sequence creates 4 RTCPeerConnections. You might want to put in a comment that the two need to be independent so that they both get the same independent limit (50k).